docs/FacialRecognitionSystem.md

```diff
@@ -0,0 +1,104 @@
+    - The dot product layer outputs the dot product representing the similarity between the embedding from the given image and embeddings in the database.
+
+Figure 1 depicts the facial recognition system sequential diagram.
+
+<img src="facialrecognition.png" style="zoom: 50%;" />
+
+​Figure 1. MAX7800x facial recognition system
+
+## Dataset
+
+The first step will be the dataset preparation. The dataset is VGGFace-2 [1].
+Please review the term and conditions at [VGGFace2](https://www.robots.ox.ac.uk/~vgg/data/vgg_face2/). Then, download the dataset and extract raw images to the train and test subfolders.
+
+Expected folder structure:
+
+                            - root_dir
+                                - VGGFace-2
+                                    - train
+                                    - test
+
+FaceID and Face Detection tasks share the same ground truth pickle, and it will be automatically generated when one of these tasks started.
+
+## Face Detection
+
+To be able to localize faces in a facial recognition system, a face detection algorithm is generally used in facial recognition systems. Face detection is an object detection problem that has various solutions in the literature. In this work, a face detection algorithm that will run on MAX7800x series microcontrollers with a real-time performance was targeted.
+
+For digit detection problem, previously, a Tiny SSD[2] based MAX7800x object detection algorithm was developed , named Tinier SSD. The face detection model is a modified version of the digit detection model. The modification was realized to reduce the number of parameters and enable larger input size.
+
+To train the facedetection model, "scripts/train_facedet_tinierssd.sh" script can be used.
+
+## FaceID
+
+To train a faceID model for MAX7800x microcontrollers, there are multiple steps. As the MAX7800x FaceID models will be trained in a knowledge distillation fashion, the first step will be downloading a backbone checkpoint for the teacher model.
+
+The suggested teacher model is IR-152, but the other teacher models defined in "model_irse_DRL.py" may be used as well. Please review the terms and conditions at face.evoLVe[3] repository, and download the checkpoint according to your teacher model selection.
+
+There are two FaceID models, one for the MAX78000 and one for the MAX78002. The MAX78000 one is named faceid_112, and it is a relatively lightweight model. To enable more performance on MAX78002, a more complex model was developed, which is named mobilefacenet_112. To train the FaceID models, "scripts/train_faceid_112.sh" and "scripts/train_mobilefacenet_112.sh" scripts can be used. Training scripts will realize Dimensionality Reduction and Relation Based-Knowledge Knowledge Distillation steps automatically. A summary of Dimensionality Reduction and Relation-Based Knowledge Distillation can be found in the following sub-sections.

```

Comment:
To train the FaceID models, "scripts/train_faceid_112.sh" and "scripts/train_mobilefacenet_112.sh" scripts can be used, respectivey.